src/extension/providers/nostr/index.ts

```diff
@@ -30,23 +42,31 @@ export default class NostrProvider {
   }
 
   async getPublicKey() {
-    await this.enable();
-    return await this.execute("getPublicKeyOrPrompt");
+    return this.queue.add(async () => {
+      await this.enable();
+      return await this.execute("getPublicKeyOrPrompt");
+    });
   }
 
   async signEvent(event: Event) {
-    await this.enable();
-    return this.execute("signEventOrPrompt", { event });
+    return this.queue.add(async () => {
+      await this.enable();
+      return this.execute("signEventOrPrompt", { event });
+    });
   }
 
   async signSchnorr(sigHash: string) {
-    await this.enable();
-    return this.execute("signSchnorrOrPrompt", { sigHash });
+    return this.queue.add(async () => {
+      await this.enable();
+      return this.execute("signSchnorrOrPrompt", { sigHash });
+    });
   }
 
   async getRelays() {
-    await this.enable();
-    return this.execute("getRelays");
+    return this.queue.add(async () => {
+      await this.enable();
+      return this.execute("getRelays");
+    });
```

Comment:
`enable` in the on and off methods should also be wrapped in the queue, right?

Reply:
For this reason I am wondering if the execute method itself should handle the queue, hmm..

Reply:
I was wondering the same. we always call execute, could execute implement what is currently in the `PromiseQueue #add`

Reply:
Yes, but maybe we could even move it to execute<T>(), I'll check.